exchange/bidder.go

```
@@ -479,7 +488,22 @@ func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.Re
 	span.SetAttributes(semconv.HTTPAttributesFromHTTPStatusCode(httpResp.StatusCode)...)
 	span.SetStatus(semconv.SpanStatusFromHTTPStatusCode(httpResp.StatusCode))
 
-	respBody, err := ioutil.ReadAll(httpResp.Body)
+	// to know if the response is gzip
+	span.SetAttributes(respCompressionKey.Bool(!httpResp.Uncompressed))
+
+	// content-length from response header
+	respHeaderContentLength := httpResp.Header.Get("content-length")
+	// content-encoding from response header
+	respHeaderContentEncoding := httpResp.Header.Get("content-encoding")
+	span.SetAttributes(respHeadersContentLengthCompressedKey.String(respHeaderContentLength))
+	span.SetAttributes(respHeadersContentEncodingKey.String(respHeaderContentEncoding))
+
+	// response body content length
+	respBody, _ := ioutil.ReadAll(httpResp.Body)
```

Why did you start ignoring err here?

There is an err check, I guess you should be fetching length after handling error